src/ng/directive/input.js

```diff
@@ -1563,6 +1565,49 @@ function rangeInputType(scope, element, attr, ctrl, $sniffer, $browser) {
     } :
     originalRender;
 
+  if (minAttrType) {
+    ctrl.$validators.min = minAttrType === 'min' && supportsRange ?
+      function noopMinValidator() {
+        // Since all browsers set the input to a valid value, we don't need to check validity
+        return true;
+      } :
+      // ngMin doesn't set the min attr, so the browser doesn't adjust the input value as setting min would
+      function minValidator(modelValue, viewValue) {
+        return ctrl.$isEmpty(viewValue) || isUndefined(minVal) || viewValue >= minVal;
+      };
+
+    setInitialValueAndObserver(minAttrType, 'min', minChange);
+  }
+
+  if (maxAttrType) {
+    ctrl.$validators.max = maxAttrType === 'max' && supportsRange ?
+      function noopMaxValidator() {
+        // Since all browsers set the input to a valid value, we don't need to check validity
+        return true;
+      } :
+      // ngMax doesn't set the max attr, so the browser doesn't adjust the input value as setting max would
+      function maxValidator(modelValue, viewValue) {
+        return ctrl.$isEmpty(viewValue) || isUndefined(maxVal) || viewValue <= maxVal;
+      };
+
+    setInitialValueAndObserver(maxAttrType, 'max', maxChange);
+  }
+
+  function setInitialValueAndObserver(attrType, attrName, changeFn) {
+    var initialAttrValue;
+    if (attrType === attrName) {
+      initialAttrValue = attr[attrName];
+      // Set the actual element attribute so that the browser can adjust the value based on
+      // the max value, which might be interpolated
+      element.attr(attrName, initialAttrValue);
+    }
+
+    // This initalizes the min/max value, so that non-support browsers validate with the correct
+    // values during the initial $render
+    changeFn(initialAttrValue);
```

Comment:
At this point we really only need to set the element attribute if we have min / max and the browser supports range. I've removed the call to changeFn as its use isn't tested currently. 
The actual issue is that all validators that are based on observation if interpolation or ngX attributes only set their values after a digest. That means when the input renders these validators' valuea aren't defined. When the observers fire we will validate again with the correct values but this causes the model to be unset if the value is invalid. Imo this should still count as initialization so that shouldn't happen. The other thing is that we validate too often (once for every observer). 
I'll open an issue for part 1 of this.